src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -131,6 +131,9 @@ public static ISet<T> ToSet<T>(this IEnumerable<T> source)
             return source as ISet<T> ?? new HashSet<T>(source);
         }
 
+        public static IReadOnlyCollection<T> ToCollection<T>(this IEnumerable<T> sequence)
+            => (sequence is IReadOnlyCollection<T> collection) ? collection : sequence.ToList();
```

Comment:
Is the implication of this helper that it's going to make a copy? Because the caller could be "surprised" that this sometimes is just returning a different pointer to the same object and so they could mutate together.

Reply:
No, the purpose is to avoid multiple enumerations.